executor/cook/io_helper.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+"""This module ensures atomic writes to stdout and stderr.
+"""
+
+import logging
+import os
+import sys
+from threading import Lock, Thread
+
+__stdout_lock__ = Lock()
+__stderr_lock__ = Lock()
+
+
+def print_out(string_data, flush=False, newline=True):
+    """Wrapper function that prints to stdout in a thread-safe manner using the __stdout_lock__ lock.
+
+    Parameters
+    ----------
+    string_data: string
+        The string to output
+    flush: boolean
+        Flag determining whether to trigger a sys.stdout.flush()
+    newline: boolean
+        Flag determining whether to output a newline at the end
+
+    Returns
+    -------
+    Nothing.
+    """
+    message = '{}{}'.format(string_data, os.linesep) if newline else string_data
+    with __stdout_lock__:
+        sys.stdout.write(message)
+    if flush:
+        sys.stdout.flush()
+
+
+def print_and_log(string_data, flush=False, newline=True):
+    """Wrapper function that prints to stdout in a thread-safe manner ensuring newline at the start.
+    The function also outputs the same message via logging.info().
+
+    Parameters
+    ----------
+    string_data: string
+        The string to output
+    flush: boolean
+        Flag determining whether to trigger a sys.stdout.flush()
+    newline: boolean
+        Flag determining whether to output a newline at the end
+
+    Returns
+    -------
+    Nothing.
+    """
+    print_out('{}{}'.format(os.linesep, string_data), flush=flush, newline=newline)
+    logging.info(string_data)
+
+
+def print_err(string_data, flush=False, newline=True):
+    """Wrapper function that prints to stderr in a thread-safe manner using the __stderr_lock__ lock.
+
+    Parameters
+    ----------
+    string_data: string
+        The string to output
+    flush: boolean
+        Flag determining whether to trigger a sys.stderr.flush()
+    newline: boolean
+        Flag determining whether to output a newline at the end
+
+    Returns
+    -------
+    Nothing.
+    """
+    message = '{}{}'.format(string_data, os.linesep) if newline else string_data
```

Comment:
Would it be better to `sys.stderr.write(os.linesep) if newline`, instead of building a string every time?

Reply:
Yes. Changed.